File: Leetcode/2294 - Partition Array Such That Maximum Difference Is K/solution.ts
function partitionArray(nums: number[], k: number): number {
    nums.sort((a, b) => a - b);
    let totalSubSeq: number = 1;
    let currMin: number = nums[0];
    for (let i: number = 1; i < nums.length; i++) {
        if (nums[i] - currMin > k) {
            totalSubSeq++;
            currMin = nums[i];
        }
    }
    return totalSubSeq;
};

File: Leetcode/2294 - Partition Array Such That Maximum Difference Is K/solution.py
def partitionArray(self, nums: List[int], k: int) -> int:
    nums.sort()
    totalSubSeq: int = 1
    currMin: int = nums[0]

    for i in range(1, len(nums)):
        if nums[i] - currMin > k:
            totalSubSeq += 1
            currMin = nums[i]

    return totalSubSeq